commons/src/main/java/com/simpletools/calculator/commons/helpers/CalculatorImpl.kt

```diff
@@ -58,21 +67,31 @@ class CalculatorImpl(calculator: Calculator) {
     }
 
     fun handleEquals() {
-        val operation = OperationFactory.forId(operator!!, firstNumber, secondNumber)
-
-        if (operation != null && (digits > 0 || operation is UnaryOperation)) {
-            setAllClear()
-            resetValues()
-            firstNumber = operation.getResult()
-            setValue(Formatter.doubleToString(firstNumber))
-            setFormula(operation.getFormula())
-            lastIsOperation = false
+        val operation : Operation?
+        if (operator != "") {   // Handle new operation
+            operation = OperationFactory.forId(operator!!, firstNumber, secondNumber)
+
+            if ((operation != null) && (digits > 0 || operation is UnaryOperation)) {
+                if (operation is UnaryOperation) {
+
```

Comment:
why not check if operation is not a unary operation (remove the blank if and just evaluate the else case)

Reply:
Will fix this. I didn't have internet at the time and couldn't remember the negation syntax for kotlin.

Reply:
Agreed